scipy/integrate/_odepackmodule.c

```diff
@@ -828,6 +828,10 @@ PyInit__odepack(void)
     PyObject *m, *d, *s;
 
     m = PyModule_Create(&moduledef);
+    if (!m) {
```

Comment:
Strictly speaking, I think you need to check every single python api call instead of just checking `PyErr_Occurred` once at the end.

Reply:
Yes, that's true technically. I didn't do that since it seems to be very common practice not to in initialization routines, and in practice it's robust enough (as argued in the linked SO post). We can change to checking every call, happy to do that since it is probably better practice. In that case I'd prefer to clean up a bunch of things, like the useless `__version__` attributes that are added to `_fitpack`, `_odepack` etc. - that's all just dead code, and the version numbers aren't even meaningful (they're just made up as far as I can tell).

Reply:
On second thought I agree, let's do this the 100% correct way. I'll update this PR.

Reply:
I pushed a change for `_odepackmodule.c`. If that looks good I'll do the rest in the same way.

Reply:
The new changes LGTM.